### DISCUSSION TITLE
Examples for CMS

### DISCUSSION THREAD

**Discussion by martinbavio** [remix-run/remix#1064] [Proposals]
These days, adding an `examples` folder to the main repo is a pretty popular and efficient strategy for adoption, which helps by reducing friction for newcomers. It's being maintained and curated in many popular projects like [Next.js](https://github.com/vercel/next.js/tree/canary/examples), [Astro](https://github.com/withastro/astro/tree/main/examples) and [Nuxt](https://github.com/nuxt/nuxt.js/tree/dev/examples). In addition, **Stackblitz** offers an [amazingly fast way](https://developer.stackblitz.com/docs/platform/importing-projects/#import-from-github) to run any example into their platform.

I believe it would be incredibly beneficial for this project to expand the already existing examples folder. I'm particularly interested in the headless CMS-related examples because I believe those are quick ways to connect Remix with the real world.

As a way to quickstart the effort, here's a list of some of the most popular headless CMS, in no particular order of importance:

- [ ] [Sanity](https://www.sanity.io/)
- [ ] [DatoCMS](https://www.datocms.com/)
- [ ] [Contentful](https://www.contentful.com)
- [ ] [Plasmic](https://www.plasmic.app)
- [ ] [Prismic](https://prismic.io)
- [ ] [Ghost](https://ghost.org/)
- [ ] [Wordpress](https://wordpress.com/)
- [ ] [Netlify CMS](https://www.netlifycms.org/)
- [ ] [Strapi](https://strapi.io/)
- [ ] [Storyblok](https://www.storyblok.com)
- [ ] [Takeshape](https://www.takeshape.io)
- [ ] [Tina](https://tina.io)
- [ ] [Keystone](https://keystonejs.com)
- [ ] [GraphCMS](https://graphcms.com)
- [ ] [Craft CMS](https://craftcms.com)
- [ ] [Agility CMS](https://agilitycms.com)
- [ ] [Butter CMS](https://buttercms.com)
- [ ] [Cosmic](https://www.cosmicjs.com)

It would be nice to stay consistent feature-based among all the implementations, in order to present a cohesive and fair group of examples, so that the companies behind each CMS could use on their own DevRel related material. We could probably join forces with them in order to build those examples together.

I'm particularly interested in knowing @kentcdodds' thoughts about this proposal and get the ball rolling. I can provide examples for some of those, but I figured it was better to start a conversation first, for organizational reasons.

**Comment by kentcdodds** [comment]
I'm definitely interested in this. I've never used any of these tools so I may not be the best one to create one of these. But I'd be happy to provide feedback on them.

One thing I'd say is that we don't need to show *everything* that you can do with these platforms and Remix. We want to keep it pretty basic so there aren't a lot of distractions for people who are just trying to get an idea of how to wire these things together. So let's limit abstractions that hide the wiring please :)

**Reply by martinbavio** [reply]
I was thinking about a basic blog example for each one of those, with as few abstractions as possible. Thoughts?

**Reply by kentcdodds** [reply]
That sounds great to me 👍

**Reply by sergiodxa** [reply]
A suggestion, they should all build the exact same app, same styles, same routes, the only difference should be te integration with the CMS layer, maybe it would be useful to have shared components between them

**Reply by martinbavio** [reply]
@sergiodxa that's what I meant when I said `It would be nice to stay consistent feature-based among all the implementations`. I'll definitely take care that the outcomes are all similar, so they are helpful for the community but also for the companies behind those products.

**Reply by tigerabrodi** [reply]
I love the idea, we use Contentful at work, and I know CMSes are being used so much :sweat_smile:

**Comment by hollandThomas** [comment]
I just opened a PR with a Strapi example. Let me know what you think!

https://github.com/remix-run/remix/pull/1388
